azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/utils/TailingDebouncer.java

```diff
@@ -30,8 +30,10 @@ public synchronized void debounce() {
         this.timer = Observable.timer(this.delay, TimeUnit.MILLISECONDS)
             .subscribeOn(Schedulers.io())
             .subscribe(ignore -> {
-                this.debounced.run();
-                this.timer = null;
+                synchronized (TailingDebouncer.this) {
+                    this.debounced.run();
```

Comment:
this line should be moved out of the `synchronized` block, 
it would, otherwise, make the debounced func run sequentially.